neovim/plugin/decorators.py

```diff
@@ -80,18 +85,22 @@ def dec(f):
 
         f._nvim_rpc_spec = {
             'type': 'command',
-            'name': name,
+            'name': command_name,
             'sync': rpc_sync,
             'opts': opts
         }
         return f
     return dec
 
 
-def autocmd(name, pattern='*', sync=False, allow_nested=False, eval=None):
+def autocmd(name=None, pattern='*', sync=False, allow_nested=False, eval=None):
```

Comment:
It can be a concern that a plugin potentially can have multiple implementations of the autocmd with disjoint patterns. At least there should be a warning in the docs. (python doesn't give a warning on method shadowing, though we could hack it with a metaclass).

Reply:
I wasn't sure about the autocmd, it doesn't have much sense here I guess.

Reply:
Maybe the parameter should be named `event`, that's why I was confused here.